server/src/services/completion/onCompletion.ts

```diff
@@ -38,44 +39,41 @@ export const onCompletion = (serverState: ServerState) => {
 
     logger.trace("onCompletion");
 
-    return serverState.telemetry.trackTiming(
-      "onCompletion",
-      async (transaction) => {
-        const { found, errorMessage, documentAnalyzer, document } =
-          await applyEditToDocumentAnalyzer(
-            serverState,
-            params.textDocument.uri,
-            (doc) => resolveChangedDocText(params, doc)
-          );
-
-        if (!found || !documentAnalyzer || !document) {
-          logger.trace(
-            `Error editing and analyzing doc within onCompletion: ${errorMessage}`
-          );
-
-          return { status: "failed_precondition", result: null };
-        }
-
-        const project = documentAnalyzer.project;
-
-        const projCtx: ProjectContext = {
-          project,
-          solFileIndex: serverState.solFileIndex,
-        };
-
-        addFrameworkTag(transaction, project);
-        const completions = doComplete(
-          documentAnalyzer,
-          params.position,
-          params.context,
-          projCtx,
+    return serverState.telemetry.trackTiming("onCompletion", async () => {
+      const { found, errorMessage, documentAnalyzer, document } =
+        await applyEditToDocumentAnalyzer(
           serverState,
-          document
+          params.textDocument.uri,
+          (doc) => resolveChangedDocText(params, doc)
+        );
+
+      if (!found || !documentAnalyzer || !document) {
+        logger.trace(
+          `Error editing and analyzing doc within onCompletion: ${errorMessage}`
         );
 
-        return { status: "ok", result: completions };
+        return { status: INTERNAL_ERROR, result: null };
```

Comment:
Why not keep it as a `FAILED_PRECONDITION`?

Reply:
Good catch,  this looks like a mistake